### PR TITLE
T23831

### PR DESCRIPTION
**Pull request by cfoch**
https://phabricator.endlessm.com/T23831

### CODE REVIEW

File: lib/installedtest.pm [hunk 1]
```diff
@@ -50,6 +50,7 @@ sub exit_user_console {
     # terminal TTY for the next user_console() call is not a no-op (and hence
     # causes the display to refresh).
     console_user_exit();
+    sleep 4;
```
Comment (manuq) [inline lib/installedtest.pm]: Why is this blocking delay needed before calling `send_key`? There is another one right after calling `send_key`. If it's needed, please add a comment with an explanation. Other delays in this file have a comment like "There's a timing problem when...".

Comment (manuq) [reply lib/installedtest.pm]: Is this the sporadic failure you were getting? If so, can the other `sleep` be increased from 4 to 8? Or has to be another `sleep` right before the `send_key`?

File: tests/desktop_dinosaurs.pm [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+sub run {
+    my $self = shift;
+    my $sdk_nightly_repo = 'http://endlessm.github.io/eos-knowledge-lib/eos-sdk-nightly.flatpakrepo';
+
+    check_desktop_clean();
+    type_very_safely("gnome-terminal\n");
+    assert_screen('desktop_terminal', 5);
+
+    type_string("flatpak remote-add --from eos-sdk-nightly " . $sdk_nightly_repo . "\n");
```
Comment (manuq) [inline tests/desktop_dinosaurs.pm]: I see these lines duplicated in the three added tests. They are the same with the exception of the app ID. So this code should be generalized in one or more functions, with app ID as parameter. https://github.com/endlessm/eos-openqa-tests/blob/dd770a9490ce4b3fafcd5054ca379ddd6f700547/tests/desktop_dinosaurs.pm#L14-L28

**Comment by cfoch** [comment]
I am reusing code, but I am using other function names because a matter of tastes. 
I thought initially to launch the apps from TTY, so "I" wouldn't have to type "gnome-terminal" in the desktop, then opening the gnome-terminal to launch the app. I tried that: "DISPLAY=:0 flatpak run ...". The problem was that launching it in this way had a different behavior. In this way the scrollbar got hidden. But launching it from the gnome-terminal doesn't show this behavior. The scrollbar is always there.

**Comment by manuq** [comment]
Yes, as you found, apps shouldn't be launched from TTY or they would get the default GTK+, not the Endless custom one. The code reuse looks great! My question for the `sleep 4` still stands. Is it needed? Or the `sleep` that follows it can be increased instead?

**Comment by pwithnall** [comment]
Any progress on this, @cfoch?

**Pull request merged**